### PR TITLE
Flaky: Sort array to avoid flaky spec when comparing values

### LINKED ISSUE #10622: [Flaky] `spec/services/order_cycle_form_spec.rb:403`
## Description
```
Failures:

  1) OrderCycleForm distributor order cycle updating shipping methods and it's valid submitter is a distributor can update his own shipping methods saves the changes
     Failure/Error:
       expect{ form.save }.to change{
         order_cycle.distributor_shipping_methods.pluck(:id)
       }.from([
                distributor_shipping_method, distributor_shipping_method2
              ]).to([distributor_shipping_method])
     
       expected `order_cycle.distributor_shipping_methods.pluck(:id)` to have initially been [49, 50], but was [50, 49]
     # ./spec/services/order_cycle_form_spec.rb:412:in `block (7 levels) in <top (required)>'

Finished in 2 minutes 12.6 seconds (files took 8.89 seconds to load)
153 examples, 1 failure

Failed examples:

rspec ./spec/services/order_cycle_form_spec.rb:[40](https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/4531201111/jobs/7981076226?pr=10621#step:10:41)3 # OrderCycleForm distributor order cycle updating shipping methods and it's valid submitter is a distributor can update his own shipping methods saves the changes
```

Context: https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/4531201111/jobs/7981076226?pr=10621

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

- Closes #10622

#### What should we test?
-  Green build
- Run spec N times? `script/rspec-slow-repeat 10 spec/services/order_cycle_form_spec.rb:403`

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

**Comment by jibees** [comment]
Thanks @filipefurtad0 !
I've cherry picked commit in this PR: 0a91f3d392

File: spec/services/order_cycle_form_spec.rb [hunk 1]
```diff
@@ -362,8 +362,10 @@
               order_cycle.coordinator.users.first
             )
 
-            expect{ form.save }.to change{ order_cycle.distributor_shipping_methods.pluck(:id) }
-              .from([distributor_shipping_method, distributor_shipping_method2])
+            expect{ form.save }.to change{
+                                     order_cycle.distributor_shipping_methods.pluck(:id).sort
+                                   }
+              .from([distributor_shipping_method, distributor_shipping_method2].sort)
```
Comment (mkllnk) [inline spec/services/order_cycle_form_spec.rb:368]: We don't actually need to sort our own array but it doesn't hurt either.

**Review by mkllnk — APPROVED** [review]

**Comment by mkllnk** [comment]
I ran it 20 times and it always passed. I'll merge this because it's an improvement.

**Comment by filipefurtad0** [comment]
Seems we've missed other cases, another occurrence here:
https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/4602506740/jobs/8131544137

**Pull request merged**